src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.vb

```diff
@@ -607,7 +607,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
             End Sub
 
             Public Function Compare(x As ISymbol, y As ISymbol) As Integer Implements IComparer(Of ISymbol).Compare
-                If x Is y Then
+                If ReferenceEquals(x, y) Then
```

Comment:
>ReferenceEquals(x, y) [](start = 19, length = 21)

Please revert

Reply:
This cannot be reverted without adding a warning suppression unless I hard code a new rule for this in the analyzer. Do you have a suggestion for a new global rule that would apply here?

Reply:
> This cannot be reverted without adding a warning suppression unless I hard code a new rule for this in the analyzer. Do you have a suggestion for a new global rule that would apply here?

```Is``` and ```IsNot``` operators in VB are explicitly reference comparison. There is nothing wrong in using them and I think analyzer should not complain about them at all. There is no confusion like in C# where ```==``` sometimes means value equality and sometimes means reference equality depending on the types involved.

---
In reply to: [280900178](https://github.com/dotnet/roslyn/pull/35466#discussion_r280900178) [](ancestors = 280900178)

Reply:
@AlekseyTs The analyzer was not created because reference equality is disallowed or bad. The analyzer was created because reference equality returns the wrong answer and has caused bugs in code operating on symbols. I was able to special case nearly every scenario used by the compiler's VB code (`is` will not warn for _concrete types_ where `=` is also allowed), but this one is the outlier.

Reply:
The reference equality returns exactly the answer it is asked for. The user explicitly requested reference equality and there is no good reason to bother him/her with any warnings about that. Any API and any operation can be misused due to confusion. However, there is no confusion in VB because there is an explicit operator that is dedicated to reference equality. Anyone can make a mistake by using ```+``` operator instead of a ```-``` operator, but for some reason we are not creating analyzers to warn on their usages.

---
In reply to: [280926428](https://github.com/dotnet/roslyn/pull/35466#discussion_r280926428) [](ancestors = 280926428)

Reply:
There is no difference between `ReferenceEquals` and `Is` here. If one satisfies the analyzer then so should the other.